### PR TITLE
fix(#2157): use stdpath cache for my_on_attach.lua

### LINKED ISSUE #2157: :NvimTreeGenerateOnAttach On Windows Not Functioning
### Discussed in https://github.com/nvim-tree/nvim-tree.lua/discussions/2155

<div type='discussions-op-text'>

<sup>Originally posted by **Nazaroni** April 22, 2023</sup>
When I try to generate ":NvimTreeGenerateOnAttach", by default it uses "/tmp/my_on_attach.lua" file path... but I'm on Windows 10 😅.
Is it possible to generate this file on Windows?</div>

### PR DESCRIPTION
**Pull request by alex-courtis**
fixes #2157

### CODE REVIEW

**Review by gegoune — APPROVED** [review]

File: doc/nvim-tree-lua.txt [hunk 1]
```diff
@@ -1219,7 +1220,8 @@ In the event of a problem please disable the experiment and raise an issue.
 Configuration for diagnostic logging.
 
     *nvim-tree.log.enable*
-    Enable logging to a file `$XDG_CACHE_HOME/nvim/nvim-tree.log`
+    Enable logging to a file `nvim-tree.log` in |stdpath| `"cache"`, usually
```
Comment (gegoune) [inline doc/nvim-tree-lua.txt:1223]: Will follow up with change to `log` location instead of `cache` for logs.

File: doc/nvim-tree-lua.txt [hunk 2]
```diff
@@ -194,9 +194,10 @@ Requirements
 
 *:NvimTreeGenerateOnAttach*
 
-    Creates and opens a new file `/tmp/my_on_attach.lua` containing an
-    |nvim-tree.on_attach| function based on your |nvim-tree.view.mappings|,
-    |nvim-tree.remove_keymaps| as well as the defaults.
+    Creates and opens a new file `my_on_attach.lua` in |stdpath| `"cache"`,
```
Comment (gegoune) [inline doc/nvim-tree-lua.txt:197]: Should we put it in `run` instead?

Per dos:
```
		run          String  Run directory: temporary, local storage
				     for sockets, named pipes, etc.
``

File: lua/nvim-tree/keymap-legacy.lua [hunk 3]
```diff
@@ -401,7 +402,7 @@ function M.cmd_generate_on_attach()
     return
   end
 
-  local name = "/tmp/my_on_attach.lua"
+  local name = utils.path_join { vim.fn.stdpath "cache", "my_on_attach.lua" }
```
Comment (gegoune) [inline lua/nvim-tree/keymap-legacy.lua:405]: Retrospectively could have named that file `nvim-tree-on-attach.lua` or something like that. :)

File: doc/nvim-tree-lua.txt (hunk 1 above)
Comment (alex-courtis) [reply doc/nvim-tree-lua.txt:1223]: Yes! cache is a bit odd... there must be a better place...

File: doc/nvim-tree-lua.txt (hunk 2 above)
Comment (alex-courtis) [reply doc/nvim-tree-lua.txt:197]: `$XDG_RUNTIME_DIR` `/run/user/1000` 

Looks like it should be sane on windows however I'd rather not take the risk...

File: lua/nvim-tree/keymap-legacy.lua (hunk 3 above)
Comment (alex-courtis) [reply lua/nvim-tree/keymap-legacy.lua]: Done. Updating wiki after merge.

File: doc/nvim-tree-lua.txt (hunk 1 above)
Comment (alex-courtis) [reply doc/nvim-tree-lua.txt:1223]: #2162

**Pull request merged**